### ISSUE TITLE
Allow empty constructur call in expected class declaration

### ISSUE THREAD

**Issue by vanniktech** [pinterest/ktlint#2425]
Using 1.0.1 the following gets flagged:

```kotlin
@OptIn(ExperimentalMultiplatform::class)
@OptionalExpectation
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.BINARY)
expect annotation class Parcelize()

@OptIn(ExperimentalMultiplatform::class)
@OptionalExpectation
@Target(AnnotationTarget.PROPERTY)
@Retention(AnnotationRetention.SOURCE)
expect annotation class IgnoredOnParcel()

@OptIn(ExperimentalMultiplatform::class)
@OptionalExpectation
@Retention(AnnotationRetention.SOURCE)
@Repeatable
@Target(AnnotationTarget.CLASS, AnnotationTarget.PROPERTY)
expect annotation class TypeParceler<T, P : Parceler<in T>>()
```

with:

```
commonParcelableApi.kt:11:34: No parenthesis expected (standard:class-signature)
commonParcelableApi.kt:17:40: No parenthesis expected (standard:class-signature)
commonParcelableApi.kt:24:60: No parenthesis expected (standard:class-signature)
```

And the formatting removes the `()` however this causes a compilation error: `e: Color.kt:35:2 This class does not have a constructor`

You can use https://github.com/vanniktech/ui/pull/100 to repro

**Comment by paul-dingemans** [comment]
The problem seems to be caused by the `expect`. I am unfamiliar with it. I have found https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/expect.html. Can you give more information why and how it is used?

**Comment by TWiStErRob** [comment]
It's not kotlin-test expect, but KMM's `expect` and `actual`.

https://kotlinlang.org/docs/multiplatform-expect-actual.html

TL;DR roughly:
 * `expect` is like an interface declaration of API to be compiled against in common code
 * `actual` is the implementation to be linked at compile time of platform-specific code

**Comment by paul-dingemans** [comment]
> It's not kotlin-test expect, but KMM's `expect` and `actual`.
> 
> https://kotlinlang.org/docs/multiplatform-expect-actual.html
> 
> TL;DR roughly:
> 
> * `expect` is like an interface declaration of API to be compiled against in common code
> * `actual` is the implementation to be linked at compile time of platform-specific code

Tnx. That is clarifying. If you stuck for now, you can either disable the `class-signature` rule entirely, or with a file annotation just only for the file in which the `expect` declaration is used.

**Comment by vanniktech** [comment]
Yup I already disabled the rule entirely. Luckily that one had no dependencies that I needed to disable as well, so all good!
